octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/OpenDetailTabEntityMouseListener.java

```diff
@@ -1,72 +1,107 @@
+import org.eclipse.core.runtime.Status;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+
+import com.hpe.adm.nga.sdk.model.EntityModel;
+import com.hpe.adm.octane.ideplugins.services.EntityService;
+import com.hpe.adm.octane.ideplugins.services.filtering.Entity;
+import com.hpe.adm.octane.ideplugins.services.mywork.MyWorkUtil;
+import com.hpe.adm.octane.ideplugins.services.util.Util;
+import com.hpe.octane.ideplugins.eclipse.Activator;
+import com.hpe.octane.ideplugins.eclipse.ui.editor.EntityModelEditor;
+import com.hpe.octane.ideplugins.eclipse.ui.editor.EntityModelEditorInput;
+import com.hpe.octane.ideplugins.eclipse.ui.entitylist.EntityMouseListener;
+
+public class OpenDetailTabEntityMouseListener implements EntityMouseListener {
+	
+	private static final Set<Entity> whitelistParentDetails = new HashSet<Entity>();
+	static {
+		whitelistParentDetails.add(Entity.USER_STORY);
+		whitelistParentDetails.add(Entity.DEFECT);
+		whitelistParentDetails.add(Entity.TASK);
+		whitelistParentDetails.add(Entity.QUALITY_STORY);
+        whitelistParentDetails.add(Entity.MANUAL_TEST);
+        whitelistParentDetails.add(Entity.GHERKIN_TEST);
+        whitelistParentDetails.add(Entity.MANUAL_TEST_RUN);
+        whitelistParentDetails.add(Entity.TEST_SUITE_RUN);
+	}
+	
+    private static final ILog logger = Activator.getDefault().getLog();
+    private static EntityService entityService = Activator.getInstance(EntityService.class);
+    
+	private EntityModel parentEntityModel;
+	private IWorkbenchPage page;
+	private IWorkbenchWindow win;
+	private IWorkbench wb;
+	
+	private void openInTab() {
+        wb = PlatformUI.getWorkbench();
+        win = wb.getActiveWorkbenchWindow();
+        page = win.getActivePage();
+	}
+
+    @Override
+    public void mouseClick(EntityModel entityModel, MouseEvent event) {
+        // Open detail tab
+        if (event.count == 2) {
+            if (Entity.USER_ITEM == Entity.getEntityType(entityModel)) {
+            	openInTab();
+                entityModel = MyWorkUtil.getEntityModelFromUserItem(entityModel);
+            }
+
+            if (Entity.COMMENT == Entity.getEntityType(entityModel)) {
+            	// Get parent info
+    			parentEntityModel = (EntityModel) Util.getContainerItemForCommentModel(entityModel).getValue();
+    			Entity parentEntity = Entity.getEntityType(parentEntityModel);
+    			if (whitelistParentDetails.contains(parentEntity)) { 
+    				openInTab();
+    				entityModel = (EntityModel) Util.getContainerItemForCommentModel(entityModel).getValue();
+    			} else {
+    				entityService.openInBrowser(entityModel);
+    			}
+            }
+            
+            if(Entity.FEATURE == Entity.getEntityType(entityModel) || Entity.EPIC == Entity.getEntityType(entityModel)) {
```

Comment:
@vladbenchea Do you think we can simplify this method using just the 'whitelistParentDetails' and the fact that the entity we just double-clicked is a comment or not? There seems to be a lot of duplicated functionality here

Reply:
@paul-ciobanu made use of our white list (now named supportedEntitiesThatAllowDetailView).